itest/lnd_route_blinding_test.go

```diff
@@ -1510,3 +1510,83 @@ func testBlindedPaymentHTLCReForward(ht *lntest.HarnessTest) {
 		require.Fail(ht, "timeout waiting for sending payment")
 	}
 }
+
+// testGivenBlindedRouteInvoices tests lnd's ability to create a blinded
+// payment path when the income node is given. Which it then inserted into
+// an invoice, sent to an invoice with a blinded path and forwarded payment
+// in a blinded route and finally, receiving the payment.
+func testGivenBlindedRouteInvoices(ht *lntest.HarnessTest) {
+	var (
+		// Convenience aliases.
+		alice = ht.Alice
+		bob   = ht.Bob
+	)
+
+	// Setup a two hop channel network: Alice -- Bob -- Carol.
+	// We set our proportional fee for these channels to zero, so that
+	// our calculations are easier. This is okay, because we're not testing
+	// the basic mechanics of pathfinding in this test.
+	chanAmt := btcutil.Amount(100000)
+	chanPointAliceBob := ht.OpenChannel(
+		alice, bob, lntest.OpenChannelParams{
+			Amt:        chanAmt,
+			BaseFee:    10000,
+			FeeRate:    0,
+			UseBaseFee: true,
+			UseFeeRate: true,
+		},
+	)
+
+	carol := ht.NewNode("Carol", nil)
+	ht.EnsureConnected(bob, carol)
+
+	var bobCarolBase uint64 = 2000
+	chanPointBobCarol := ht.OpenChannel(
+		bob, carol, lntest.OpenChannelParams{
+			Amt:        chanAmt,
+			BaseFee:    bobCarolBase,
+			FeeRate:    0,
+			UseBaseFee: true,
+			UseFeeRate: true,
+		},
+	)
+
+	// Wait for Alice to see Bob/Carol's channel because she'll need it for
+	// pathfinding.
+	ht.AssertTopologyChannelOpen(alice, chanPointBobCarol)
+
+	// Lookup full channel info so that we have channel ids for our route.
+	bobCarolChan := ht.GetChannelByChanPoint(bob, chanPointBobCarol)
+
+	// Let Carol add a blinded invoice choosing the income channel.
+	var (
+		minNumRealHops      uint32 = 1
+		numHops             uint32 = 1
+		incomingChannelList        = []uint64{bobCarolChan.ChanId}
+	)
+
+	invoice := carol.RPC.AddInvoice(&lnrpc.Invoice{
+		Memo:      "test",
+		ValueMsat: 10_000_000,
+		IsBlinded: true,
+		BlindedPathConfig: &lnrpc.BlindedPathConfig{
+			MinNumRealHops:      &minNumRealHops,
+			NumHops:             &numHops,
+			IncomingChannelList: incomingChannelList,
```

Comment:
can you also test the case where a non existing channel id causes no route to be created ?

Reply:
I'm testing it ... but the AddInvoice trigger the rpc error: code = Unknown, desc = could not find any routes to self to use for blinded route construction.

And I'm not finding a way of getting the error to assert it! Any Tip?

Reply:
I think comparing the text is good enough, we don't have structured errors yet for RPC

Reply:
I`m not finding a way, I got the following message:
```
    harness_rpc.go:100: 
                Error Trace:    /home/pins-dev/Projects/lnd/lntest/rpc/harness_rpc.go:100
                                                        /home/pins-dev/Projects/lnd/lntest/rpc/lnd.go:290
                                                        /home/pins-dev/Projects/lnd/itest/lnd_route_blinding_test.go:1538
                                                        /home/pins-dev/Projects/lnd/lntest/harness.go:387
                                                        /home/pins-dev/Projects/lnd/itest/lnd_test.go:139
                Error:          Received unexpected error:
                                rpc error: code = Unknown desc = could not find any routes to self to use for blinded route construction
                Messages:       Node"C": failed to call AddInvoice
```

And cannot take the error message to compare with a text!!

Am I missing something?

Reply:
you need to add this in the harness:

```
// AddInvoiceAssertErr makes a RPC call to AddInvoice and asserts an error
// has returned.
func (h *HarnessRPC) AddInvoiceAssertErr(req *lnrpc.Invoice) error {
	ctxt, cancel := context.WithTimeout(h.runCtx, DefaultTimeout)
	defer cancel()

	_, err := h.LN.AddInvoice(ctxt, req)
	require.Error(h, err, "expect add invoice to fail")

	return err
}
```

Reply:
and instead of `AddInvoice` call `AddInvoiceAssertErr`